src/ParseObject.js

```diff
@@ -2335,8 +2335,10 @@ const DefaultController = {
       });
 
     } else if (target instanceof ParseObject) {
-      // copying target lets Flow guarantee the pointer isn't modified elsewhere
+      // generate _localId in case if cascadeSave=false
+      target._getId();
```

Comment:
Maybe we should check if it is cascadeSave and singleInstance = false to avoid calling this method when not necessary? What do you think?

Reply:
I think it's better call it once anyway to make sure the _localId is generated. Many things depend on it and so it seems wrong that it's only generated as a side effect of other calls (like _getStateIdentifier or unsavedChildren)